sdk/eventhub/azure-eventhub/tests/perfstress_tests/send_event_batch.py

```diff
@@ -0,0 +1,38 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ._test_base import _SendTest
+
+from azure_devtools.perfstress_tests import get_random_bytes
+
+from azure.eventhub import EventData
+
+
+class SendEventBatchTest(_SendTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.data = get_random_bytes(self.args.event_size)
+        self.to_send_event_data_batches = []
+        self._preload_event_data_batches()
+
+    def _preload_event_data_batches(self):
+        batch = self.producer.create_batch()
+        for _ in range(self.args.num_events):
+            try:
+                batch.add(EventData(self.data))
+            except ValueError:
+                # Batch full
+                self.to_send_event_data_batches.append(batch)
+                batch = self.producer.create_batch()
+                batch.add(EventData(self.data))
+        self.to_send_event_data_batches.append(batch)
```

Comment:
We should make this part of the actual run. It's part of the code that a user would have to write, and they would be unlikely to separate out batch creation from sending like this - so our test should reflect that.

We may need to do a `globa_setup` that "starts" the connection so that when the first `create_batch` method is called we have already got the supported batch size.

Reply:
updated code to have an explicit call `creat_batch` in `gloab_setup` which will communicate with the service and cache the max allowed link frame size, also updated sending part.